### ISSUE TITLE
Enum argument can be null in Java, but exhaustive when contains no null branch

### ISSUE THREAD

**Issue by pambrose** [mscheong01/krotoDC#2]
First, thanks so much for this project. It should be part of the Kotlin/gRPC distro.

The generated `.toDataClass()` methods are generating this compiler warning
```
Enum argument can be null in Java, but exhaustive when contains no null branch
```
for the when argument. Adding 
```
null -> null
```
would get rid of the warning. 

Again, thanks for your efforts with this project.

**Comment by mscheong01** [comment]
Hi, thanks a lot for filing this issue 🙂 
I was actually aware of this warning, and it seems a bit weird that they come up since the java enum fields are marked with `@Nullable`. Maybe it's a small crack in the Kotlin - Java interoperability.
However, I agree that we should avoid having warnings in the generated code if possible. I'll add the null -> clause to genereated when statements and have the warnings removed in the next patch.

**Comment by mscheong01** [comment]
This was released as [1.0.1](https://central.sonatype.com/namespace/io.github.mscheong01) thanks again!

**Comment by pambrose** [comment]
Looks good! Thanks.
